packages/carousel/src/Carousel/Carousel.tsx

```diff
@@ -10,30 +10,34 @@ import {
   SlideProps,
 } from './types';
 
-export const Carousel: CarouselProps = ({ children, showDots = true, showArrows = false, ...props }) => {
+export const Carousel: CarouselProps = ({ children, showDots = true, showArrows = false, visibleSlides, ...props }) => {
   const [totalSlides] = Children.map(children, (child) => {
     if (isValidElement(child) && child.type === Carousel.Slider) {
       return Children.count(child.props.children) as number;
     }
   });
   if (!totalSlides) return;
 
-  const renderDots = (totalSlides: number) => {
+  const renderDots = (totalSlides: number, visibleSlides: number) => {
     const carouselDots: ReactNode[] = [];
-    for (let i = 0; i < totalSlides; i++) {
-      carouselDots.push(<Carousel.Dot key={i} slide={i} />);
+    const totalSlideGroups = Math.ceil(totalSlides / visibleSlides);
+
+    for (let i = 0; i < totalSlideGroups; i++) {
+      const slide = i * visibleSlides;
+      carouselDots.push(<Carousel.Dot key={slide} slide={slide} data-testid="carousel-dot" />);
```

Comment:
Could we replace data-testid in favor of aria-label?

Reply:
updated